sw/device/lib/testing/test_framework/ottf_console.c

```diff
@@ -63,6 +68,37 @@ void *ottf_console_get(void) {
   }
 }
 
+static status_t uart_getc(void *io) {
+  const dif_uart_t *uart = (const dif_uart_t *)io;
+  uint8_t byte;
+  TRY(dif_uart_byte_receive_polled(uart, &byte));
+  TRY(ottf_console_flow_control(uart, kOttfConsoleFlowControlAuto));
+  return OK_STATUS(byte);
+}
+
+/*
+ * The user of this function needs to be aware of the following:
+ * 1. The exact amount of data expected to be sent from the device side must be
+ * known in advance.
+ * 2. Characters should be retrieved from the console as soon as they become
+ * available. Failure to do so may result in an SPI transaction timeout.
+ */
+static status_t spi_device_getc(void *io) {
+  static size_t index = 0;
+  static upload_info_t info = {0};
+
+  if (index == info.data_len) {
+    memset(&info, 0, sizeof(upload_info_t));
+    CHECK_STATUS_OK(
+        spi_device_testutils_wait_for_upload(&ottf_console_spi_device, &info));
```

Comment:
I think instead of `&ottf_console_spi_device` here you want to cast the `io` pointer to the spi_device DIF handle, like is done for the `uart_getc` function.

Reply:
Okay, done.